tox.ini

```diff
@@ -1,25 +1,27 @@
 [tox]
-envlist = 
-    py27,py36,py37,py38,py39,pypy,cover
+envlist =
+    py27,py36,py37,py38,py39,pypy27,pypy3,coverage
 
 [testenv]
-commands = 
-    python setup.py dev
-    python setup.py test
-deps =
-    py27: pyramid<=2.0
-    py27: mock<=4.0
-    py27: waitress<=2.0
+extras =
+    tests
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
+commands =
+    python --version
+    coverage run -m pytest {posargs:}
 
-[testenv:cover]
-basepython =
-    python3.9
-commands = 
-    python setup.py dev 
-    python setup.py nosetests
+[testenv:coverage]
+skip_install = true
+commands =
+    -coverage combine
+    -coverage xml
```

Comment:
Usage of `-` here allows the command to fail but continue. The last `coverage report` will fail if anything happens.

This means:

 * You get to see the report regardless of the outcome of the XML step
 * You can run this more than once if the combine step has already removed the source files